### PR TITLE
ETL-2989: Add file level retention for fork operator

### PR DESCRIPTION
**Pull request by ydailinkedin**
1. Added ProxyableFileLevelDatasetProfile for finding file level datasets with proxy enabled.
2. Added SingleTimestampVersionFinder, which uses TimestampedDatasetVersion to find single versions.
3. Added FileListUtils, which is moved from internal idpc repo.

### CODE REVIEW

**Comment by ydailinkedin** [comment]
@ibuenros Can you review?

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/profile/ProxyableFileLevelDatasetProfile.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.data.management.retention.dataset.ConfigurableDataset;
+import gobblin.data.management.retention.dataset.Dataset;
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.util.FileListUtils;
+import gobblin.util.ProxiedFileSystemCache;
+
+
+public class ProxyableFileLevelDatasetProfile extends ConfigurableGlobDatasetFinder {
```
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/profile/ProxyableFileLevelDatasetProfile.java]: Please add javadoc

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/SingleTimestampVersionFinder.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+import gobblin.data.management.retention.dataset.Dataset;
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.TimestampedDatasetVersion;
+
+
+public class SingleTimestampVersionFinder implements VersionFinder<TimestampedDatasetVersion> {
```
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/SingleTimestampVersionFinder.java]: javadoc

**Comment by ibuenros** [comment]
This is just a suggestion:
Why not make a single dataset per glob match, and then make each file under the dataset a TimeStampedDatasetVersion? For example, if the glob is /user/*, each user will be a dataset, with one DatasetVersion per file owned by that user. You can do the proxying in the VersionFinder. Advantages:
* No need to modify ConfigurableGlobDatasetFinder
* Get empty directory cleanup for free
* (Debatable) cleaner implementation

**Comment by ydailinkedin** [comment]
@ibuenros Updated.

**Comment by ibuenros** [comment]
LGTM

**Pull request merged**